core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/AvroUpsertMonitor.scala

```diff
@@ -194,63 +219,116 @@ class AvroUpsertMonitorActor(
-          IO.fromFuture(IO(workspaceService.apply(petUserInfo).batchUpdateEntities(WorkspaceName(avroMetadataJson.namespace, avroMetadataJson.name), upsertBatch, true)))
-        }.unsafeToFuture
+          IO.fromFuture(IO(ws.batchUpdateEntities(workspaceName, upsertBatch, true)))
+        }.unsafeToFuture}
     } yield {
-      logger.info(s"completed Avro upsert job ${avroMetadataJson.jobId} for user: ${avroMetadataJson.userEmail} with ${avroUpsertJson.size} entities")
+      logger.info(s"completed async upsert job ${jobId} for user: ${userInfo.userEmail} with ${upsertJson.size} entities")
       ()
     }
   }
 
-  private def writeResult(path: String, message: String): Future[Unit] = {
-    logger.info(s"writing import result to $path")
-    googleStorage.createBlob(GcsBucketName(avroUpsertBucketName), GcsBlobName(path), message.getBytes).compile.drain.unsafeToFuture()
-  }
-
-  private def resultString(message: String) = s"""{\"message\":\"${message}\"}"""
 
   private def acknowledgeMessage(ackId: String) = {
     logger.info(s"acking message with ackId $ackId")
     pubSubDao.acknowledgeMessagesById(pubSubSubscriptionName, Seq(ackId))
   }
 
+  case class AvroUpsertAttributes(workspace: WorkspaceName, userEmail: RawlsUserEmail, importId: UUID, upsertFile: String)
+
   private def parseMessage(message: PubSubMessage) = {
-    message.contents.parseJson.asJsObject.getFields("name").head.compactPrint match {
-      case objectIdPattern(jobId, file) => (jobId, file)
-      case m => throw new Exception(s"unable to parse message $m")
-    }
+    val workspaceNamespace = "workspaceNamespace"
+    val workspaceName = "workspaceName"
+    val userEmail = "userEmail"
+    val jobId = "jobId"
+    val upsertFile = "upsertFile"
+
+    def attributeNotFoundException(attribute: String) =  throw new Exception(s"unable to parse message - attribute $attribute not found in ${message.attributes}")
+
+    AvroUpsertAttributes(
+      WorkspaceName(message.attributes.getOrElse(workspaceNamespace, attributeNotFoundException(workspaceNamespace)),
+        message.attributes.getOrElse(workspaceName, attributeNotFoundException(workspaceName))),
+      RawlsUserEmail(message.attributes.getOrElse(userEmail, attributeNotFoundException(userEmail))),
+      UUID.fromString(message.attributes.getOrElse(jobId, attributeNotFoundException(jobId))),
+      message.attributes.getOrElse(upsertFile, attributeNotFoundException(upsertFile))
+    )
   }
 
-  private def readUpsertObject(path: String): Future[Seq[EntityUpdateDefinition]] = {
-    readObject[Seq[EntityUpdateDefinition]](path, decompress = true)
-  }
 
-  private def readMetadataObject(path: String): Future[AvroMetadataJson] = {
-    readObject[AvroMetadataJson](path)
+  private def readUpsertObject(path: String) = {
+    val pathArray = path.split("/")
+    val bucketName = GcsBucketName(pathArray(0))
+    val blobName = GcsBlobName(pathArray(1))
```

Comment:
this would fail if we the path array has more than two parts.  Maybe  `GcsBlobName(pathArray.tail.mkString("/"))` ... or would we need `GcsBlobName(pathArray.tail.mkString("%2f"))`?